### PR TITLE
Stop using zerolog/log.Logger and use logger from context instead

### PR DESCRIPTION
**Pull request by michel-laterman**
## What is the problem this PR solves?

I tried to change the json schema generation tool to a version that supports serializing to `time.Time`, however there were a lot of test-failures.
It was very difficult to track why tests were failing because a lot of fleet-server's codebase (like the bulker) outputs to zerolog's global logger, while other sections use embedded (or per-request) loggers and the output of tests is muddled and inconsistent to read.
Trying to change the global logger for a test case introduces a race-condition failure within the test.
If we use an alternate method of getting a "global" logger that we can adjust for tests it would make reading test output much easier.

## How does this PR solve the problem?

Stop using the global zerolog logger and use one that is passed through the context instead.
Set context default to the same as the global-logger (the one that is controlled through the fleet-server's `logger` package) so that when fleet-server is running normally there is no change, however it is much easier to change for test cases so we can have log output that's tied to the specific test making it easier to track any errors in tests.
Change how the fleet-server logger package reloads config when adjusting for new output

## How to test this PR locally

```sh
make test
```

### CODE REVIEW

**Review by michel-laterman — COMMENTED** [review]
This is a very tedious pr where I mostly just replaced `log.` with `zerolog.Ctx(ctx).`

**Review by kpollich — APPROVED** [review]
This makes sense to me, and thanks for dealing with the tedium of replacing or updating all function signatures to accept the new logger context. We do this (passing a `logger` around) all over Kibana too, so I'm pretty comfortable with this pattern - especially since it makes writing tests/assertions for specific logging cases much easier. LGTM 🚀

File: internal/pkg/action/dispatcher.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func (d *Dispatcher) Subscribe(agentID string, seqNo sqn.SeqNo) *Sub {
 	sz := len(d.subs)
 	d.mx.Unlock()
 
-	log.Trace().Str(logger.AgentID, agentID).Int("sz", sz).Msg("Subscribed to action dispatcher")
+	zerolog.Ctx(context.TODO()).Trace().Str(logger.AgentID, agentID).Int("sz", sz).Msg("Subscribed to action dispatcher")
```
[the remark this answers is not in the document]
Comment (nchaulet) [reply internal/pkg/action/dispatcher.go:87]: should we create a follow up issue for adding a context to these functions?

**Review by nchaulet — COMMENTED** [review]
LGTM 🚀

**Comment by michel-laterman** [comment]
This pr was to replace all global refs. We still pass `zerolog.Logger` objects around in some function signatures; these should at least now come from the context.
If needed we can create a follow up issue if we really want to remove the explicit logger passes in our codebase (I know this is a common occurrence in the api package)

File: internal/pkg/action/dispatcher.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michel-laterman) [reply internal/pkg/action/dispatcher.go:87]: https://github.com/elastic/fleet-server/issues/3087

**Pull request merged**